### DISCUSSION TITLE
KeePassXC: How can I auto-fill but NOT auto-submit?

### DISCUSSION THREAD

**Discussion by watchpocket** [keepassxreboot/keepassxc#7185] [Q&A]
First off, I'm using **KeePassXC** (not _KeePass2_ or _KeyPassX_ or _KeePassDX_) on a desktop workstation.

I regularly visit a website where I want to be able to **_auto-fill_** both the username & password credential boxes, but I don't want to **_auto-submit_** the username & password.

You can't have auto-submit on a login page where, when you log out, you're returned instantly to the login page again.

If auto-submit is activated on a page that sends you **_back_** to the login page when you log off, the auto-submit will kick in again after you log off and after you then get returned to the login page, so that you'll never be able to log out.

(Though yes, the website will eventually log you off after some amount of time that you're inactive on the site.)

In KeePassXC, in the "Edit entry" --> "Auto-Type" window for this particular webpages' entry, I've checked "Enable Auto-Type for this entry" and "Use custom Auto-Type sequence."

In the box below that, I've changed

{USERNAME}{TAB}{PASSWORD}{ENTER}

to just:

{USERNAME}{TAB}{PASSWORD}

thinking that leaving out "{ENTER}" would prevent the actual log-in, until if and when I manually hit the <ENTER> key.

**_But the auto-submit takes place anyway._**

(I never had this problem when I was using KeePass2, by the way. Or rather, it was more straightforward in that program what to do about it.)

So, my question:

**_What do I need to do to have the login take place only when I hit <ENTER>?_**

I'm using:

KeePassXC Version: 2.6.6

KeePassXC-Browser Version: 1.7.10.1

Firefox Version: 95.0b12

Ubuntu Linux 20.04.3, kernel 5.11.0-40-generic

(P.S. -- In the _KeePassXC-Browser_ extension's Settings -> General page, there is a warning about halfway down (in the "filling Credentials" section) that says: "_Warning! Using auto-fill is not safe. Use at your own risk."_

But I think it's **_auto-submit_** that's dangerous, not auto-fill, exactly for the reason I state above.)

**Comment by droidmonkey** [comment]
Use either Browser Extension OR Auto-Type  but not both. Make sure you didn't define a custom sequence in a window association that includes the enter key. 

Autofill is dangerous on browser extension because your credentials could be harvested by a malicious script on the page that holds bogus login fields. Auto fill removes the human interaction aspect which is what makes it dangerous.
